taskcoordinator/app/src/test/java/com/microservices/taskcoordinator/controllers/CoordinatorControllerIntegrationTests.java

```diff
@@ -77,12 +85,113 @@ public void testCoordinateOrder_allValid_orderCoordinatedAndLaundryStateChanged(
         OrderDto orderCreated = orderService.getOrderById(INPUT_ORDER_ID);
         LaundryStateDto laundryStateForCreatedOrder = laundryStateService.getLaundryStateById(orderSubmissionDto.getLaundryId());
 
-        Assert.assertEquals(coordinationDto.getOrderId(), orderSubmissionDto.getOrderId());
-        Assert.assertEquals(orderSubmissionDto.getOrderId(), orderCreated.getId());
-        Assert.assertEquals(OrderStatus.RESERVED, orderCreated.getStatus());
+        assertEquals(coordinationDto.getOrderId(), orderSubmissionDto.getOrderId());
+        assertEquals(orderSubmissionDto.getOrderId(), orderCreated.getId());
+        assertEquals(OrderStatus.RESERVED, orderCreated.getStatus());
 
         long reserveDiff = laundryStateForCreatedOrder.getReservedTime() - laundryWillBeChosen.getReservedTime();
-        Assert.assertEquals((long) orderCreated.getDuration(), reserveDiff);
+        assertEquals((long) orderCreated.getDuration(), reserveDiff);
     }
 
+    @Test
+    @Sql(scripts = "/test-data/one_laundry.sql")
+    @Transactional
+    public void testSubmitOrder_fromReservedStatus() throws Exception {
+        int orderId = 1;
+        int laundryId = 1;
+
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(laundryId, 200L, 2);
+        OrderSubmittedDto orderSubmittedDto = new OrderSubmittedDto(orderId, inboundLaundryStateDto);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        mockMvc.perform(put("/orders/666/status/submitted")
```

Comment:
666? really?

Reply:
I just wanted to make sure that that number doesn't affect anything

Reply:
you've just pointed out on a api's flaw. if so, please, change service code to validate consistency of dto's and url's id